internal/yggdrasil/yggdrasil.go

```diff
@@ -416,3 +432,134 @@ func (h *Handler) setStorageConfiguration(ctx context.Context,
 
 	return err
 }
+
+func (h *Handler) readAndValidateSecret(ctx context.Context, secretName, secretNamespace string, secretKeys keyMapType) (*corev1.Secret, error) {
```

Comment:
Can we have those secret manipulation functions moved to some `secrets.go` module?

Reply:
these are specific to yggdrasil.go. It's just for making the functions smaller.
Let's wait till we see how we deal with other resources and then we can have a 'pod spec parsing' package

Reply:
I would create another package for it too.

Reply:
@machacekondra when u get to configMaps, if you see fit to put all these functions in a single package then ping me and I'll do it for you.